apps/vite/src/components/calendar/EventList.tsx

```
@@ -1,26 +1,71 @@
-          <UpcomingEvent key={index} event={event} isHoliday={isHoliday} />
-        ))}
+        {isHoliday && (
+          <>
+            {newEventDetails
+              .filter(
+                (event) =>
+                  event.isHoliday &&
+                  !isBefore(new Date(event.enDate), new Date())
+              )
+              .map((event, index) => (
+                <UpcomingEvent key={index} event={event} isHoliday />
```

Biome

[error] Avoid using the index of an array as key property in an element.

This is the source of the key value.

The order of the items may change, and this also affects performances and component state.
Check the React documentation. 

(lint/suspicious/noArrayIndexKey)